File: M3 Intermediate DSA/src/L7_Arrays_Subarrays/Assignment7.java
package L7_Arrays_Subarrays;

public class Assignment7 {

  public int[][] solve(int[] A) {
    int N = A.length;
    int size = N * (N + 1) / 2;
    int[][] ans = new int[size][];
    int ansIndex = 0;

    for (int index = 0; index < N; index++) {
      for (int jIndex = index; jIndex < N; jIndex++) {
        int[] subArray = new int[jIndex - index + 1];
        int subArrayIndex = 0;

        for (int kIndex = index; kIndex <= jIndex; kIndex++) {
          subArray[subArrayIndex] = A[kIndex];
          subArrayIndex++;
        }

        ans[ansIndex] = subArray;
        ansIndex++;
      }
    }

    return ans;
  }

}

File: M3_Intermediate_DSA/L5_Subarrays/assignment7.go
package main

func SolveA7(A []int) [][]int {
	N := len(A)
	size := N * (N + 1) / 2
	ans := make([][]int, size)
	ansIndex := 0

	for index := range A {
		for jIndex := index; jIndex < N; jIndex++ {

			subArr := make([]int, jIndex-index+1)
			sunArrIndex := 0

			for kIndex := index; kIndex <= jIndex; kIndex++ {
				subArr[sunArrIndex] = A[kIndex]
				sunArrIndex++
			}

			ans[ansIndex] = subArr
			ansIndex++
		}
	}

	return ans
}
